src/simple_cache_client.rs

```diff
@@ -1107,6 +1110,116 @@ impl SimpleCacheClient {
+    ///
+    /// # Example
+    /// ```
+    /// # fn main() -> anyhow::Result<()> {
+    /// # tokio_test::block_on(async {
+    /// use uuid::Uuid;
+    /// use std::num::NonZeroU64;
+    /// use std::time::Duration;
+    /// use momento::{CollectionTtl, SimpleCacheClientBuilder};
+    /// use momento::response::FetchResponse;
+    /// #   use futures::FutureExt;
+    /// #   use std::panic::AssertUnwindSafe;
+    ///
+    /// let auth_token = std::env::var("TEST_AUTH_TOKEN").expect("TEST_AUTH_TOKEN must be defined");
+    /// let cache_name = Uuid::new_v4().to_string();
+    /// let set_name = Uuid::new_v4().to_string();
+    /// let ttl = CollectionTtl::default();
+    ///
+    /// let mut momento = SimpleCacheClientBuilder::new(auth_token, Duration::from_secs(30))?
+    ///     .build();
+    ///
+    /// momento.create_cache(&cache_name).await?;
+    /// #   let result: Result<anyhow::Result<()>, _> = AssertUnwindSafe(async {
+    /// momento.set_union(&cache_name, &*set_name, vec!["a", "b", "c", "d"], ttl).await?;
+    /// momento.set_difference(&cache_name, &*set_name, vec!["b", "d"]).await?;
+    ///
+    /// let response = momento.set_fetch(&cache_name, &*set_name).await?;
+    /// let set = response.value.expect("set was missing within the cache");
+    ///
+    /// assert_eq!(set.len(), 2);
+    /// assert!(set.contains("a".as_bytes()));
+    /// assert!(set.contains("c".as_bytes()));
+    /// assert!(!set.contains("b".as_bytes()));
+    /// assert!(!set.contains("d".as_bytes()));
+    /// #       Ok(())
+    /// #   }).catch_unwind().await;
+    /// #   momento.delete_cache(&cache_name).await?;
+    /// #   match result {
+    /// #       Ok(result) => result,
+    /// #       Err(panic) => std::panic::resume_unwind(panic),
+    /// #   }
+    /// # })
+    /// # }
+    /// ```
+    pub async fn set_difference<E: IntoBytes>(
+        &mut self,
+        cache_name: &str,
+        set_name: impl IntoBytes,
+        elements: Vec<E>,
+    ) -> MomentoResult<FetchResponse<()>> {
+        use set_difference_request::subtrahend::{Set as SubtrahendSetSet, SubtrahendSet};
+        use set_difference_request::{Difference, Subtrahend};
+        use set_difference_response::Set;
+
+        let request = self.prep_request(
+            cache_name,
+            SetDifferenceRequest {
+                set_name: set_name.into_bytes(),
+                difference: Some(Difference::Subtrahend(Subtrahend {
+                    subtrahend_set: Some(SubtrahendSet::Set(SubtrahendSetSet {
+                        elements: elements.into_iter().map(|e| e.into_bytes()).collect(),
+                    })),
+                })),
+            },
+        )?;
+
+        let response = self.data_client.set_difference(request).await?.into_inner();
+
+        Ok(match response.set {
+            Some(Set::Found(_)) => FetchResponse::Found(()),
+            Some(Set::Missing(_)) | None => FetchResponse::Missing,
+        })
```

Comment:
This isn't a "fetch" response. Maybe you want to name it `SetResponse` (gross) or something? Maybe @malandis or @pgautier404 have a thought here

Reply:
My thought was that I could use this type across a bunch of different methods. I agree, though, that the name doesn't really match in this case. I'm not sure what a better one would be though.